src/Tools/dotnet-monitor/CollectionRules/Options/Actions/CollectLogsOptions.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Diagnostics;
+
+#if UNITTEST
+namespace Microsoft.Diagnostics.Monitoring.TestCommon.Options
+#else
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions
+#endif
+{
+    /// <summary>
+    /// Options for the CollectLogs action.
+    /// </summary>
+    [DebuggerDisplay("CollectLogs")]
+    internal sealed partial class CollectLogsOptions
+    {
+        [EnumDataType(typeof(LogLevel))]
+        public LogLevel? LogLevel { get; set; }
+
+        public Dictionary<string, LogLevel?> FilterSpecs { get; set; }
+
+        public bool? UseAppFilters { get; set; }
+
+        [Range(typeof(TimeSpan), ActionOptionsConstants.Duration_MinValue, ActionOptionsConstants.Duration_MaxValue)]
+        public TimeSpan? Duration { get; set; }
```

Comment:
Do you think there's any value in making some of the more common action fields inherit from a base class (I'm assuming configuration can handle this). For example, actions that have egress and duration can share from a base option.

Reply:
Configuration will handle it, however the schema.json file will largely not be usable for IntelliSense for the fields that are not on the deriving class. I'll try it again and check if that is still the case or if I messed something up.

Reply:
I'll take a look at this refactor when I go to add the action options to the JSON schema.